### ISSUE TITLE
[OSS-Fuzz] Introspector build failures since using new PM

### ISSUE THREAD

**Issue by Navidem** [ossf/fuzz-introspector#305]
Since merge of https://github.com/google/oss-fuzz/pull/7788, around 59 projects fail to build.

As I checked some including `json`, `valijson`, `wabt`, `wolfssl`, and `znc`, the error message is:

```
/usr/bin/ld.gold: fatal error: LLVM gold plugin: <unknown>:0: Undefined temporary symbol .LtmpXXXXX
```

**Comment by DavidKorczynski** [comment]
This is a duplicate of https://github.com/ossf/fuzz-introspector/issues/48

**Comment by DavidKorczynski** [comment]
When the issue was posted the build was based on the new pass manager: Marco built the project using the patches as they were in oss_fuzz_integration/oss-fuzz-patches.diff at the time, which did not include `-flegacy-...` flags (https://github.com/ossf/fuzz-introspector/blob/b9e2761b2533bea544b540e55953a3900ff1a887/oss_fuzz_integration/oss-fuzz-patches.diff#L9), meaning the new PM was used at that point. We only switched back to the old pass manager in the local fuzz-introspector set up around this PR https://github.com/ossf/fuzz-introspector/pull/141

**Comment by Navidem** [comment]
Thanks, 

> meaning the new PM was used at that point. 

So it means we ack some projects will have such issue with new PM.

**Comment by Navidem** [comment]
+1 on sticking to new PM.
Projects can enable old PM on demand if they find it necessary to build Fuzz Introspector.

**Comment by Navidem** [comment]
Should be fixed once https://github.com/google/oss-fuzz/pull/7828 is merged.

**Comment by Navidem** [comment]
> Should be fixed once [google/oss-fuzz#7828](https://github.com/google/oss-fuzz/pull/7828) is merged.

This fixed the majority. Still 18 are failing, will open another issue as the error is not about this issue anymore.
